plugins/genesys/fu-genesys-usbhub-device.c

```diff
@@ -746,10 +748,19 @@ fu_genesys_usbhub_device_setup(FuDevice *device, GError **error)
 		} else if (memcmp(self->static_ts.mask_project_ic_type, "3590", 4) == 0) {
 			self->isp_model = ISP_MODEL_HUB_GL3590;
 		} else {
-			g_set_error_literal(error,
-					    FWUPD_ERROR,
-					    FWUPD_ERROR_INTERNAL,
-					    "Unknown ISP model");
+			g_autofree gchar *ic_type =
+			    fu_common_strsafe((const gchar *)self->static_ts.mask_project_ic_type,
+					      6);
+			g_set_error(error,
+				    FWUPD_ERROR,
+				    FWUPD_ERROR_NOT_SUPPORTED,
+				    "Unsupported IC GL%c%c%c%c-%c%c",
```

Comment:
I don't think you know that strlen(ic_type) >= 6 -- I think you either want to print this as %s (probably fine given it's in the error path) or make a `convert_mask_project_ic_type_to_string` function that adds the dash in the right place.

Reply:
make sense.
done